### DISCUSSION TITLE
Using a smaller pick target to highlight a parent mesh

### DISCUSSION THREAD

**Discussion by cxreiff** [aevyrie/bevy_mod_picking#215] [Q&A] [answered]
Hello- I have a use case (highlighting a hovered tile) where I need to highlight a mesh but only when the bottom face of the target mesh is hovered over. I figured a quick hack would be to spawn in a child mesh aligned with the bottom face, add `RaycastPickTarget` to the child, and add `Highlight { hovered, pressed }` to the parent, letting the hover event bubble up to the parent level. However, from my testing, the `Highlight` component only seems to work when the event originates from the same entity, not from a child.

Does this use case force me to handle highlighting manually using `OnPointer::<Over>::send_event`, or am I misunderstanding some key part of how the `bevy_picking_highlight` crate works? Is there a way to get the default highlighting system to handle bubbled events?

For clarity, my use case:
<img width="270" alt="example" src="https://github.com/aevyrie/bevy_mod_picking/assets/8195503/9dc0c25b-de29-4a94-846e-bc724edcd3da">
I would like the transparent hex to appear when hovering over the green x, but not when hovering the red x.

Thanks for your time!

**Comment by aevyrie** [comment, marked as answer]
Highlighting looks at the `Interaction` state of an entity to determine what highlight to use (None, Clicked, Hovered).

https://github.com/aevyrie/bevy_mod_picking/blob/1f7df34a30a6bdfd4d0512bea6bc83e69ec98b11/crates/bevy_picking_highlight/src/lib.rs#L274-L276

For your use case, I think all you need to do is add an eventlistener on your parent to update its `Interaction` component. This isn't handled out of the box because there isn't an obvious way to merge the states of multiple children with conflicting `Interaction` states.

**Reply by cxreiff** [reply]
Thank you! Works as advertised.

For future reference to anybody coming across this answer, in order to handle the case where the mouse is released and the parent `Interaction` state can go to either `Hovered` or `None` depending on the mouse position, you need to pull the interaction state from the child rather than manually set it based on the event type. To do so, you should add `Interaction::None` to the child when spawning it, so that its interaction state actually gets updated (if there is no Interaction component it will not be added when the state changes).

**Reply by aevyrie** [reply]
I also just realized you have a potentially simpler option here. mod_raycast allows you to use a `SimplifiedMesh` instead of the rendered mesh for raycasting against. You could just use this, and you wouldn't need any workarounds for highlighting.

**Reply by cxreiff** [reply]
Ah- is there a way of accessing `bevy_mod_raycast` functionality through this package, or are you suggesting that I set up a `RaycastMesh`/`RaycastSource`/`DefaultRaycastingPlugin`/`SimplifiedMesh` as if I was setting up `bevy_mod_raycast` directly?

I gave the latter a shot (using a `MyRaycastSet` type parameter, following the [simplified_mesh](https://github.com/aevyrie/bevy_mod_raycast/blob/main/examples/simplified_mesh.rs) example), but the highlight didn't appear to be changing.

Also, it looks like SimplifiedMesh just takes a single `mesh`, so this will only work if the simplified mesh is in the exact same position, right (my use case is a flat mesh aligned with the bottom-most face)?

**Reply by aevyrie** [reply]
`bevy_mod_raycast` is used as a backend provider for this picking crate. The backend that is supplied out-of-the-box assumes you are using a single raycasting set `RaycastPickingSet`. To work with multiple sets, you would either need to make your own backend based off of the supplied on, or we would need to update the backend to allow specifying the set to use, which would allow users to add a backend for each set.

**Reply by cxreiff** [reply]
Ah, thanks! Glancing at the 'bevy_picking_raycast' backend, it looks a fair bit more complex than just copying the Interaction state from child to parent ([link](https://github.com/cxreiff/hex_tiles/blob/5cfa3e4ef6c6281812bec3122b4d688cd533b9f2/src/world_plugin.rs#L163)). If my use case becomes more complex I'll take another look.

Thanks for all your help!
